nucypher/characters/control/interfaces.py

```diff
@@ -189,6 +191,19 @@ def retrieve(self,
 
         self.implementer.join_policy(label=label, publisher_verifying_key=alice_verifying_key)
 
+        if self.implementer.federated_only:
+            from nucypher.policy.maps import TreasureMap as _MapClass
+        else:
+            from nucypher.policy.maps import SignedTreasureMap as _MapClass
+
+        # TODO: This LBYL is ugly and fraught with danger.  NRN
```

Comment:
Let's give the issue number of the follow-up issue. I'm pretty sure none of this code gets used because of  https://github.com/nucypher/nucypher/pull/2730#discussion_r667964421.

```suggestion
        # TODO: This LBYL is ugly and fraught with danger.  NRN - #2751
```